src/cmap/auth/mongodb_oidc/workflow.ts

```diff
@@ -8,7 +8,11 @@ export interface Workflow {
    * All device workflows must implement this method in order to get the access
    * token and then call authenticate with it.
    */
-  execute(connection: Connection, credentials: MongoCredentials): Promise<Document>;
+  execute(
+    connection: Connection,
+    credentials: MongoCredentials,
+    reauthenticate?: boolean
```

Comment:
Personal preference - I think flags are more clear than booleans.  Feel free to ignore and close this comment
```suggestion
    mode: 'authenticate' | 'reauthenticate' = 'authenticate'
```

Reply:
if we do address this though, we should make this change everywhere we use `reauthenticate` or `reauthenticating` in this PR

Reply:
I think since it's an "on" or "off" scenario I feel like the boolean is fine since the property name explains what the state is.